contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxMetrics.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.jmxmetrics;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.util.Properties;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class JmxMetrics {
+  private static final Logger logger = Logger.getLogger(JmxMetrics.class.getName());
+
+  private final Timer timer = new Timer();
+  private GroovyRunner runner;
+
+  /**
+   * Begins the core metric scraping and reporting loop on configured interval after parsing and
+   * binding the configured groovy script and establishing the {@link JmxClient} connection.
+   *
+   * @param config - {@link JmxConfig} with Groovy script path, JMX connection info, and metric
+   *     export options.
+   */
+  public void start(final JmxConfig config) {
+    JmxClient jmxClient;
+    try {
+      jmxClient = new JmxClient(config);
+    } catch (MalformedURLException e) {
+      throw new ConfigureError("Malformed serviceUrl: ", e);
+    }
+
+    runner = new GroovyRunner(config.groovyScript, jmxClient, new GroovyUtils(config));
+
+    timer.scheduleAtFixedRate(
```

Comment:
`scheduleWithFixedDelay` since it's an interval

Reply:
Let's use singlethread executor instead of timer, no need for the wrapping

Reply:
Updated